tests/integration/targets/latest/tasks/static_group_association.yml

```
@@ -0,0 +1,89 @@
+---
+##
+##
+### PYNAUTOBOT_STATIC_GROUP_ASSOCIATION
+##
+##
+
+- name: "NAUTOBOT 2.3+ STATIC GROUP ASSOCIATION TESTS"
+  when: 
+    - "nautobot_version is version('2.3', '>=')"
+  block:
+    - set_fact:
+        static_dynamic_group: "{{ lookup('networktocode.nautobot.lookup', 'dynamic-groups', api_endpoint=nautobot_url, token=nautobot_token, api_filter='name=\"TestStaticAssociations\"') }}"
+        static_device: "{{ lookup('networktocode.nautobot.lookup', 'devices', api_endpoint=nautobot_url, token=nautobot_token, api_filter='name=TestDeviceR1') }}"
+
+    - name: "STATIC GROUP ASSOCIATION 1: Necessary info creation"
+      networktocode.nautobot.static_group_association:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        dynamic_group: "TestStaticAssociations"
+        associated_object_type: "dcim.device"
+        associated_object_id: "{{ static_device['key'] }}"
+        state: present
+      register: test_one
+
+    - name: "STATIC GROUP ASSOCIATION 1: ASSERT - Necessary info creation"
+      assert:
+        that:
+          - test_one is changed
+          - test_one['diff']['before']['state'] == "absent"
+          - test_one['diff']['after']['state'] == "present"
+          - test_one['static_group_association']['dynamic_group'] == static_dynamic_group['key']
+          - test_one['static_group_association']['associated_object_type'] == "dcim.device"
+          - test_one['static_group_association']['associated_object_id'] == static_device['key']
+          - "'created' in test_one['msg']"
+
+    - name: "STATIC GROUP ASSOCIATION 2: Test duplication association (Idempotency)"
+      networktocode.nautobot.static_group_association:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        dynamic_group: "{{ static_dynamic_group['key'] }}"
```

```suggestion
        dynamic_group: "TestStaticAssociations"
```

The idempotent task should be exactly the same to ensure the first task wasn't a false positive.